alea/include/alps/alea/plugin/stream_serializer.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 1998-2018 ALPS Collaboration. See COPYRIGHT.TXT
+ * All rights reserved. Use is subject to license terms. See LICENSE.TXT
+ * For use in publications, see ACKNOWLEDGE.TXT
+ */
+#pragma once
+
+#include <numeric>
+#include <type_traits>
+
+#include <alps/alea/core.hpp>
+#include <alps/alea/complex_op.hpp>
+#include <alps/alea/mean.hpp>
+#include <alps/alea/variance.hpp>
+#include <alps/alea/covariance.hpp>
+#include <alps/alea/autocorr.hpp>
+#include <alps/alea/batch.hpp>
+
+namespace alps { namespace alea {
+
+// Metafunction to detect ALEA result types
+template<typename T> struct is_alea_result : std::false_type {};
```

Comment:
I thought we could put these 6 lines into main ALEA sources, a place like `alea/include/alps/alea/core.hpp`.
Or scatter them over `mean.hpp`, `variance.hpp` and so on...

Reply:
Hmmm, right now we have this `traits<...>` struct; this is the C++98 "way" of doing these things ... I have the feeling It has since falled out of favor somewhat with C++11 and whatnot ... do you think we shall merge it into something like` `traits<...>::IS_RESULT` or do you think it is better separated out?

Reply:
> Hmmm, right now we have this traits<...> struct; this is the C++98 "way" of doing these things ... I have the feeling It has since falled out of favor somewhat with C++11 and whatnot ...

Trait bundles similar to `traits<...>` have become less popular for a reason. When you ask for `traits<mean_result<double>>::HAVE_MEAN` you make compiler spend time on instantiating all other members of the structure too.

> do you think we shall merge it into something liketraits<...>::IS_RESULT` or do you think it is better separated out?

Calling `traits<double>::IS_RESULT` would lead to a compilation error, unless you include `IS_RESULT` into the main declaration of `struct traits`. This is not very nice, I think...